src/System.Security.AccessControl/tests/Ace.Tests.cs

```diff
@@ -0,0 +1,482 @@
+            Assert.Throws<ArgumentOutOfRangeException>("opaque", () => ObjectAce_CreateTestData(2, 1, 2, "S-1-5-11", 1, "{73D03E18-5C03-422c-9EC7-C4C5B1CB1612}", "{3CFD5DF9-9146-43c8-BF99-78E7E2DE4BAF}", true, 1, 0));
+            Assert.Throws<ArgumentOutOfRangeException>("opaque", () => ObjectAce_CreateTestData(2, 1, 2, "S-1-5-11", 1, "{73D03E18-5C03-422c-9EC7-C4C5B1CB1612}", "{3CFD5DF9-9146-43c8-BF99-78E7E2DE4BAF}", true, 17, 0));
+        }
+
+        [Fact]
+        public void ObjectAce_CreateBinaryForm_Invalid()
+        {
+            ObjectAce ace = (ObjectAce)ObjectAce_CreateTestData(0, 0, 1, "S-1-5-11", 1, "{73D03E18-5C03-422c-9EC7-C4C5B1CB1612}", "{3CFD5DF9-9146-43c8-BF99-78E7E2DE4BAF}", false, 8, 0)[0];
+            Assert.Throws<ArgumentNullException>("binaryForm", () => ObjectAce.CreateFromBinaryForm(null, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => ObjectAce.CreateFromBinaryForm(new byte[1], -1));
+            Assert.Throws<ArgumentException>("binaryForm", () => ObjectAce.CreateFromBinaryForm(new byte[ace.BinaryLength + 1], 2));
+            Assert.Throws<ArgumentException>("binaryForm", () => ObjectAce.CreateFromBinaryForm(new byte[ace.BinaryLength], 1));
+        }
+
+        [Fact]
+        public void ObjectAce_GetBinaryForm_Invalid()
+        {
+            ObjectAce ace = (ObjectAce)ObjectAce_CreateTestData(0, 0, 1, "S-1-5-11", 1, "{73D03E18-5C03-422c-9EC7-C4C5B1CB1612}", "{3CFD5DF9-9146-43c8-BF99-78E7E2DE4BAF}", false, 8, 0)[0];
+            Assert.Throws<ArgumentNullException>("binaryForm", () => ace.GetBinaryForm(null, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => ace.GetBinaryForm(new byte[1], -1));
+            Assert.Throws<ArgumentOutOfRangeException>("binaryForm", () => ace.GetBinaryForm(new byte[ace.BinaryLength + 1], 2));
+            Assert.Throws<ArgumentOutOfRangeException>("binaryForm", () => ace.GetBinaryForm(new byte[ace.BinaryLength], 1));
+        }
+
+        [Theory]
+        [MemberData(nameof(ObjectAce_TestObjects))]
+        public void ObjectAce_GetBinaryForm(GenericAce testAce, byte[] expectedBinaryForm, int testOffset)
+        {
+            byte[] resultBinaryForm = new byte[testAce.BinaryLength + testOffset];
+            testAce.GetBinaryForm(resultBinaryForm, testOffset);
+            GenericAce_VerifyBinaryForms(expectedBinaryForm, resultBinaryForm, testOffset);
+        }
+
+        [Theory]
+        [MemberData(nameof(ObjectAce_TestObjects))]
+        public void ObjectAce_CreateFromBinaryForm(GenericAce expectedAce, byte[] testBinaryForm, int testOffset)
+        {
+            GenericAce resultAce = ObjectAce.CreateFromBinaryForm(testBinaryForm, testOffset);
+            GenericAce_VerifyAces(expectedAce, resultAce);
+        }
+    }
+}
```

Comment:
Tests I didn't see:

* GenericAce.Equals
    * It looks like if you made a `TestAce : GenericAce` and it had the same binary form as another Ace type that Equals would match even though the types don't.  If that's true in core (and desktop) we should have a test to make sure we don't ever break that.
* GetHashCode
    * Shouldn't really be tested for a known value, but that two distinct instances which .Equals the same GetHashCode the same.  It's hard to codify that "different should be different" because of how hashing works, but feel free to make some sort of generalization :)
* op_Equals
    * Nulls, `a == a`, value equals
* op_NotEquals
    * Pretty easily bolted on to the previous test

After GenericAce I stopped scrolling through looking to see if there were any other public members not being utilized.